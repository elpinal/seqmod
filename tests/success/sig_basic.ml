{
  type bool = unit + unit

  signature EQ = {
    type t
    val eq : t & t -> bool
  }

  signature ORD = {
    include EQ
    val less : t & t -> bool
  }

  signature SET = {
    type set
    type elem
    val empty : set
    val add : elem & set -> set
    val mem : elem & set -> bool
  }
}
