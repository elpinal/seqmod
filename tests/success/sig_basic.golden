purity: pure
{EQ=∃!22, !23.{eq(=!23):!22 & !22 -> unit + unit, t(=!22):Type}, ORD=∃!24, !25, !26.{eq(=!25):!24 & !24 -> unit + unit, less(=!26):!24 & !24 -> unit + unit, t(=!24):Type}, SET=∃!27, !28, !29, !30, !31.{add(=!30):!28 & !27 -> !27, empty(=!29):!27, mem(=!31):!28 & !27 -> unit + unit, elem(=!28):Type, set(=!27):Type}, bool(=unit + unit):Type}
